### ISSUE TITLE
Support for azurerm_backup_policy_file_share Hourly Backups

### ISSUE THREAD

**Issue by rukas** [hashicorp/terraform-provider-azurerm#16352]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently, `azurerm_backup_policy_file_share` only supports daily backups in the provider, but the actual resource in Azure now supports hourly backups.

[Announcement from Azure](https://azure.microsoft.com/en-us/updates/azurefiles-multiplebackups/)

When I make the change in the Azure portal the terraform resource sees the change but I'm unable to set the backup frequency to `Hourly`. It only accepts `Daily` [per the resource documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/backup_policy_file_share#frequency)

### New or Affected Resource(s)/Data Source(s)

azurerm_backup_policy_file_share

### Potential Terraform Configuration

```hcl
resource "azurerm_backup_policy_file_share" "policy" {
  name                = "file-share-policy"
  resource_group_name = azurerm_resource_group.example.name
  recovery_vault_name = azurerm_recovery_services_vault.example.name

  timezone = "UTC"

  backup {
    frequency = "Hourly"
    runTimes  = [
      "00:00",
      "04:00",
      "08:00",
      "12:00",
      "16:00",
      "20:00"
    ]
  }

  retention_daily {
    count = 10
  }
}
```

### References

- https://azure.microsoft.com/en-us/updates/azurefiles-multiplebackups/

**Comment by yuanb** [comment]
I'd like to have this too!

**Comment by philthynz** [comment]
Yea this would be great to have. If anyone is interested I have deployed an Hourly policy via ARM:

```
resource "azurerm_resource_group_template_deployment" "backup_policy" {
  name                = "backup-policy-arm"
  resource_group_name = var.rg_name
  deployment_mode     = "Incremental"

  parameters_content = jsonencode({
    "name" = {
      value = "${var.recovery_vault_name}/${var.backup_policy_name}"
    }
  })

  template_content = <<TEMPLATE
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "name": {
            "type": "String"
        }
    },
    "variables": {},
    "resources": [
        {
            "type": "Microsoft.RecoveryServices/vaults/backupPolicies",
            "apiVersion": "2022-01-01",
            "name": "[parameters('name')]",
            "properties": {
                "backupManagementType": "AzureStorage",
                "retentionPolicy": {
                    "dailySchedule": {
                        "retentionDuration": {
                            "count": 14,
                            "durationType": "Days"
                        }
                    },
                    "retentionPolicyType": "LongTermRetentionPolicy",
                    "weeklySchedule": {
                        "daysOfTheWeek": [
                            "Sunday"
                        ],
                        "retentionDuration": {
                            "count": 26,
                            "durationType": "Weeks"
                        }
                    }
                },
                "schedulePolicy": {
                    "hourlySchedule": {
                        "interval": 4,
                        "scheduleWindowDuration": 10,
                        "scheduleWindowStartTime": "0001-01-01T08:00:00"
                    },
                    "schedulePolicyType": "SimpleSchedulePolicy",
                    "scheduleRunFrequency": "Hourly",
                    "scheduleWeeklyFrequency": 0
                },
                "timeZone": "New Zealand Standard Time",
                "workLoadType": "AzureFileShare"
            }
        }
    ]
}
TEMPLATE
}
```

**Comment by Netkracker** [comment]
We at @swisspost also miss the feature to directly set hourly file backup via tf ressource "azurerm_backup_policy_file_share "

any news on this ?

**Comment by katbyte** [comment]
fixed by #21529
